test/integration/tasks.test.js

```diff
@@ -293,4 +293,51 @@ describe('Tasks', function () {
         })
     })
   })
+
+  describe('PATCH /self/:id', function () {
+    const taskStatusData = {
+      status: 'currentStatus',
+      percentCompleted: 50
+    }
+
+    it('Should update the task status for given self taskid', function (done) {
+      chai
+        .request(app)
+        .patch(`/tasks/self/${taskId1}`)
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send(taskStatusData)
+        .end((err, res) => {
+          if (err) { return done(err) }
+          expect(res).to.have.status(200)
+          expect(res.body.message).to.equal('Task updated successfully!')
+          return done()
+        })
+    })
+
+    it('Should return 404 if task doesnt exist', function (done) {
+      chai
+        .request(app)
+        .patch('/tasks/self/wrongtaskId')
+        .set('cookie', `${cookieName}=${jwt}`)
+        .send(taskStatusData)
+        .end((err, res) => {
+          if (err) { return done(err) }
+          expect(res).to.have.status(404)
+          expect(res.body.message).to.equal('Task doesn\'t exist')
+          return done()
+        })
+    })
+
+    it('Should return Forbidden error if task is not assigned to self', async function () {
+      const { user: { id } } = await userModel.fetchUser({ username: 'akshay' })
```

Comment:
you'll first need to add this user in DB right?

Reply:
I don't think we need to add user in DB.